docs/src/userguide/tempita.rst

```diff
@@ -0,0 +1,295 @@
+.. _tempita:
+
+Tempita templating language
+===========================
+
+This document describes template syntax features supported by Tempita - expressions, control blocks, escaping, filters, etc.
```

Comment:
I think it might be worth adding a quick description here. Something like:

> Tempita is used internally in Cython for code-generation purposes. It is also available to users as a public API so they can generate or customize code when building a Cython module.